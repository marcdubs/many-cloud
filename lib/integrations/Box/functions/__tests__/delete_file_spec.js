describe("delete_file", () => {
  const expected_resolution = "Some file info";
  let data, func;

  beforeEach(() => {
    data = {
      client: {
        files: {
          delete: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              resolve(expected_resolution);
            });
          })
        }
      }
    };
    func = require("../delete_file")(data);
  });

  describe("when the deletion is successfull", () => {
    let actual;
    beforeEach(async () => {
      actual = await func(1234);
    });

    it("returns the expected resolution", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("when the deletion returns an error", () => {
    beforeEach(() => {
      data = {
        client: {
          files: {
            delete: jest.fn().mockImplementation(() => {
              return new Promise((resolve, reject) => {
                throw new Error(
                  "Something absolutely terrible happened. Oh no."
                );
              });
            })
          }
        }
      };
      func = require("../delete_file")(data);
    });

    it("rejects with the same error", async () => {
      await expect(func(1234)).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
