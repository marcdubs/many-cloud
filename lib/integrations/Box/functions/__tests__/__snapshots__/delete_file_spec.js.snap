// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete_file when the deletion returns an error rejects with the same error 1`] = `"Something absolutely terrible happened. Oh no."`;
